core/org.wso2.carbon.ui/src/main/resources/web/dialog/js/dialog.js

```diff
@@ -286,6 +286,102 @@ CARBON.showPopupDialog = function(message, title, windowHight, okButton, callbac
+ * @return {Boolean}
+ */
+CARBON.showPopupConfirm = function (htmlMessage, title, windowHeight, okButton, cancelButton, callback, windowWidth) {
+    if (!isHTML(htmlMessage)) {
+        htmlMessage = htmlEncode(htmlMessage);
+    }
+    var strDialog = "<div id='dialog' title='" + title + "'><div id='popupDialog'></div>" + htmlMessage + "</div>";
+    var requiredWidth = 750;
+    if (windowWidth) {
+        requiredWidth = windowWidth;
+    }
+    var func = function () {
+        jQuery("#dcontainer").html(strDialog);
+        if (okButton) {
+            jQuery("#dialog").dialog({
+                close: function () {
+                    jQuery(this).dialog('destroy').remove();
+                    jQuery("#dcontainer").empty();
+                    return false;
+                },
+                buttons: {
+                    "OK": function () {
+                        if (callback && typeof callback == "function")
+                            callback();
+                        jQuery(this).dialog("destroy").remove();
+                        jQuery("#dcontainer").empty();
+                        return false;
+                    },
+                    "Cancel": function () {
+                        jQuery(this).dialog('destroy').remove();
+                        jQuery("#dcontainer").empty();
+                        return false;
+                    },
+                },
+                height: windowHeight,
+                width: requiredWidth,
+                minHeight: windowHeight,
+                minWidth: requiredWidth,
+                modal: true
+            });
+        } else {
+            jQuery("#dialog").dialog({
+                close: function () {
+                    jQuery(this).dialog('destroy').remove();
+                    jQuery("#dcontainer").empty();
+                    return false;
+                },
+                height: windowHeight,
+                width: requiredWidth,
+                minHeight: windowHeight,
+                minWidth: requiredWidth,
+                modal: true
+            });
+        }
+
+        if (okButton) {
+            $('.ui-dialog-buttonpane button:contains(OK)').attr("id", "dialog-confirm_ok-button");
+            $('#dialog-confirm_ok-button').html(okButton);
+        }
+        if (cancelButton) {
+            $('.ui-dialog-buttonpane button:contains(Cancel)').attr("id", "dialog-confirm_cancel-button");
+            $('#dialog-confirm_cancel-button').html(cancelButton);
+        }
+
+
+        jQuery('.ui-dialog-titlebar-close').click(function () {
+            jQuery('#dialog').dialog("destroy").remove();
+            jQuery("#dcontainer").empty();
+            jQuery("#dcontainer").html('');
+        });
+
+    };
+    if (!pageLoaded) {
+        jQuery(document).ready(func);
+    } else {
+        func();
+    }
+
+    function isHTML(str) {
+        var a = document.createElement('div');
```

Comment:
Shall we use an qualified variable name?

Reply:
Improved the function with https://github.com/wso2/carbon-kernel/pull/1803/commits/6c2f3c086dfd6bafc157856ffccc3c3fea9ab0f4#diff-c10be0f2b78f6e8ff271bb8b8c4818efR375